src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -204,8 +208,34 @@ public void RegisterCommandHandler<TCommand>(Func<TCommand, KernelInvocationCont
         }
 
         public void RegisterCommandType<TCommand>()
-            where TCommand : KernelCommand =>
-            KernelCommandEnvelope.RegisterCommandTypeReplacingIfNecessary<TCommand>();
+            where TCommand : KernelCommand
+        {
+            if (_registeredCommandTypes.Add(typeof(TCommand)))
+            {
+                KernelCommandEnvelope.RegisterCommandTypeForSerialization<TCommand>();
+                var defaultHandler = CreateDefaultHandlerForCommandType<TCommand>() ?? throw new InvalidOperationException("CreateDefaultHandlerForCommandType should not return null");
```

Comment:
This still bugs me. We should differentiate the command types in a future PR.

Reply:
we should, agreed